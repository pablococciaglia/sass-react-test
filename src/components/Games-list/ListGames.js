import React from 'react';

export const ListGames = ({ games }) => {
	return (
		//function .map to set the each list item with the info from the .json
		<>
			{games.map((game) => (
				<li key={Math.random()} className='listbox__list'>
					<div className='list__flexcontainer list__background mt7'>
						<h2 className='list__textinfo'>X</h2>
						<img
							className='list__img ml7'
							src={game.url}
							alt='a good game'
						></img>

						<div className='list__textsbox'>
							<h2 className='list__textcompany'>{game.company}</h2>
							<strong className='list__texttitle'>{game.title}</strong>
							<strong className='list__textstake'>
								â‚¬{game.minbet} min. Stake
							</strong>
						</div>

						<button className='list__button'>PLAY</button>
					</div>

					<hr />
				</li>
			))}
		</>
	);
};
