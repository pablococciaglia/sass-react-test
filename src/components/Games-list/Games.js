import React, { useState, useEffect } from 'react';
import { ListGames } from './ListGames';

export const Games = ({ select }) => {
	const [games, setgames] = useState([]);
	//useState: where we will save the .json file.

	useEffect(() => {
		//useEffect: without dependency to tun it just once

		const customData = require('../../assets/games.json');
		setgames(customData);
		// we use require because the .json file is on local
	}, []);

	switch (
		select //switch: depending on the choice, the objects saved in 'games' will be sorted and then displayed
	) {
		case 'name':
			games.sort((a, b) => {
				//sort the objet array by letters
				const wordA = a.title.toLowerCase();
				const wordB = b.title.toLowerCase();
				if (wordA < wordB) {
					return -1;
				}
				if (wordA > wordB) {
					return 1;
				}
				return 0;
			});

			return (
				//render the list of components sinding the 'games' via props
				<ul>
					<ListGames games={games} />
				</ul>
			);

		case 'stakemin':
			games.sort((a, b) => {
				//sort the objet array from min to max stake
				if (a.minbet < b.minbet) {
					return -1;
				}
				if (a.minbet > b.minbet) {
					return 1;
				}
				return 0;
			});

			return (
				<ul>
					<ListGames games={games} />
				</ul>
			);

		case 'stakemax':
			games.sort((a, b) => {
				//sort the objet array from max to min stake
				if (a.minbet < b.minbet) {
					return 1;
				}
				if (a.minbet > b.minbet) {
					return -1;
				}
				return 0;
			});

			return (
				<ul>
					<ListGames games={games} />
				</ul>
			);

		default:
			return (
				//the list will be display in the original order that is in the .json
				<ul>
					<ListGames games={games} />
				</ul>
			);
	}
};
