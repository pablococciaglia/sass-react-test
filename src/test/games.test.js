import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';
import { Games } from '../components/Games-list/Games';

const wrapper = shallow(<Games />);

describe('test on Games.js', () => {
     
    test('create a snapshot of the component', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('show if when we start it brings the .json file as an object', async () => {
        const customData = require('../assets/games.json')
        expect (typeof customData).toBe('object');
    })

});
 