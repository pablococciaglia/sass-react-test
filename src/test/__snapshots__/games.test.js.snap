// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on Games.js create a snapshot of the component 1`] = `
<ul>
  <ListGames
    games={Array []}
  />
</ul>
`;
